presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/HdfsParquetDataSource.java

```diff
@@ -220,19 +227,41 @@ public void free()
             mergedRangeLoader.free();
         }
 
-        Map<K, ChunkReader> sliceStreams = slices.build();
+        Multimap<K, ChunkReader> sliceStreams = slices.build();
         verify(sliceStreams.keySet().equals(diskRanges.keySet()));
         return sliceStreams;
     }
 
-    private <K> Map<K, ChunkReader> readLargeDiskRanges(Map<K, DiskRange> diskRanges)
+    @Override
+    public ColumnIndex readColumnIndex(ColumnChunkMetaData column) throws IOException
```

Comment:
I can move these methods, but I need to be able to call the Parquet library method Util.readColumnIndex(inputStream) to read the column index. Should I expose the underlying InputStream by adding a ParquetDataSource.getInputStream() method?

Reply:
You can wrap the Slice returned by ChunkReader with an InputStream to pass to that method

Reply:
I could change it to this instead:

    public ColumnIndex readColumnIndex(int offset, PrimitiveType type)

Reply:
No, what I meant is that you can move these methods elsewhere so that this class has no knowledge about ColumnIndex or OffsetIndex. The caller would take the ChunkReader, get a Slice, wrap it with an InputStream and pass it to the `Util.readColumnIndex(...)` method.

Reply:
See my comment here: https://github.com/trinodb/trino/pull/7349/files#r665764776